src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/subscribe_button_response.cc

```diff
@@ -55,6 +56,38 @@ SubscribeButtonResponse::~SubscribeButtonResponse() {}
 
 void SubscribeButtonResponse::Run() {
   SDL_LOG_AUTO_TRACE();
+
+  int32_t corr_id = (*message_)
+                        .getElement(strings::params)
+                        .getElement(strings::correlation_id)
+                        .asInt();
+
+  hmi_apis::Common_Result::eType hmi_result =
+      static_cast<hmi_apis::Common_Result::eType>(
+          (*message_)
+              .getElement(strings::params)
+              .getElement(hmi_response::code)
+              .asInt());
+
+  if (application_manager_.IsButtonRequestExpired(corr_id)) {
+    if (!CommandImpl::IsHMIResultSuccess(
+            hmi_result, HmiInterfaces::HMI_INTERFACE_Buttons)) {
+      application_manager_.DeleteExpiredButtonRequest(corr_id);
```

Comment:
@ychernysheva it should be deleted from collection even if the result was not successful because we are looking at the fact that an expired response has been received. Suggest to move this line outside the if statement and assign app_id and button_name to separate constants before deleting

Reply:
Done in 950d7d4